packages/shared/lib/contexts/onboarding/actions/initialiseOnboardingProfile.ts

```diff
@@ -1,16 +1,30 @@
-import { NetworkProtocol } from '@core/network'
-import { profileManager } from '@core/profile-manager'
 import { get } from 'svelte/store'
+
+import { stopPollingLedgerNanoStatus } from '@core/ledger'
+import { NetworkProtocol } from '@core/network'
+import { destroyProfileManager, profileManager } from '@core/profile-manager'
+
 import { OnboardingProfileManagerAlreadyInitializedError } from '../errors'
 import { buildOnboardingProfile } from '../helpers'
-import { onboardingProfile } from '../stores'
+import { isOnboardingLedgerProfile, onboardingProfile } from '../stores'
 
 /**
  * Builds a new onboarding profile and sets the Svelte store accordingly.
  */
-export function initialiseOnboardingProfile(isDeveloperProfile: boolean, networkProtocol?: NetworkProtocol): void {
+export function initialiseOnboardingProfile(
+    isDeveloperProfile: boolean,
+    networkProtocol?: NetworkProtocol,
+    overrideExistingProfileManager = false
```

Comment:
is there a case where we wouldn't want to override the old manager?

Reply:
There are some spots in the onboarding where this is called without overriding the existing manager. It would be worth going through onboarding and cleaning this logic up a bit more from the first refactoring at some point.